app/sensors/CategorySearch.js

```diff
@@ -446,11 +446,14 @@ export default class CategorySearch extends Component {
 				value: this.state.currentValue.label
 			});
 		}
+		if (this.props.onBlur) this.props.onBlur(event);
 	}
 
 	handleKeyPress(event) {
 		if (event.key === "Enter") {
 			event.target.blur();
+		} else {
```

Comment:
Why do we have an `else` block here? Any good reasons?

Reply:
I thought the other way around. But, yeah they should always be executed when present.